### PR TITLE
Feature/grammar checker

### PR DESCRIPTION
**Pull request by jlunenburg**
Merge after https://github.com/tue-robotics/tue-env-targets/pull/93 and https://github.com/tue-robotics/action_server/pull/119

### CODE REVIEW

File: action_server/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.swp
```
Comment (MatthijsBurgh) [inline action_server/.gitignore]: Move to `.gitignore` in main folder of the repo?

Comment (jlunenburg) [reply action_server/.gitignore]: See 05430a5743e1ad00c23fd8ba128ffb5a33ddeb0a

**Comment by jlunenburg** [comment]
We'll wait until https://github.com/tue-robotics/tue_robocup/pull/888 has been merged

File: action_server/src/action_server/test_tools.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        assert "actions" in actions_definition, "'actions' not in actions definition, don't know what to do"
+
+        # Test the action description with the task manager
+        try:
+            test_result = task_manager.set_up_state_machine(
+                recipe=actions_definition["actions"],
+            )  # type: ConfigurationResult
+        except Exception as e:
+            test_result = ConfigurationResult()
+            test_result.message = "Configuration crashed: {}".format(e.message)
```
Comment (MatthijsBurgh) [inline action_server/src/action_server/test_tools.py]: Add full traceback. As current messages are only last line and don't give a clue about where it actually happened.

File: action_server/src/action_server/test_tools.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        elif test_result.config_result.message and test_result.config_result.missing_field:
+            print("Configuration of '{}' did not succeed due to missing information ({}): {}".format(
+                action, test_result.config_result.missing_field, test_result.config_result.message
+            ))
+        else:
+            failed_test_results[action] = test_result
+
+    error_str = "\n"
+    for action, test_result in failed_test_results.iteritems():
+        error_str += "\nConfiguration of action '{}' failed.\n\tRecipe: {}\n\tError: {}".format(
```
Comment (MatthijsBurgh) [inline action_server/src/action_server/test_tools.py]: Is is needed to start with a newline, when new items are also added with a new line?

**Pull request merged**